From 173808bd30234317391a13f4e0f80ac10c22854a Mon Sep 17 00:00:00 2001
From: Jose Alamos <jose@alamos.cc>
Date: Mon, 5 Jul 2021 15:45:56 +0200
Subject: [PATCH] expose dsme_atomic.h

---
 helper/DSMEAtomic.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/helper/DSMEAtomic.h b/helper/DSMEAtomic.h
index 1e862c1..1be3f14 100644
--- a/helper/DSMEAtomic.h
+++ b/helper/DSMEAtomic.h
@@ -43,7 +43,7 @@
 #ifndef DSMEATOMIC_H_
 #define DSMEATOMIC_H_
 
-#include "../../dsme_atomic.h"
+#include "opendsme/dsme_atomic.h"
 
 namespace dsme {
 
-- 
2.32.0

