From bf1ef3c529260a5809949d89b3af24257880fc28 Mon Sep 17 00:00:00 2001
From: Jose Alamos <jose@alamos.cc>
Date: Mon, 5 Jul 2021 15:51:52 +0200
Subject: [PATCH] expose DSMEMessage.h

---
 dsmeLayer/gtsManager/GTSManager.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dsmeLayer/gtsManager/GTSManager.h b/dsmeLayer/gtsManager/GTSManager.h
index 8813ffc..083e8c1 100644
--- a/dsmeLayer/gtsManager/GTSManager.h
+++ b/dsmeLayer/gtsManager/GTSManager.h
@@ -43,7 +43,7 @@
 #ifndef GTSMANAGER_H_
 #define GTSMANAGER_H_
 
-#include "../../../DSMEMessage.h"
+#include "opendsme/DSMEMessage.h"
 #include "../../helper/DSMEBufferedMultiFSM.h"
 #include "../../helper/DSMEFSM.h"
 #include "../../helper/Integers.h"
-- 
2.32.0

