From 7d071b32c2ee2e64e8682c39ba33c77969d3182d Mon Sep 17 00:00:00 2001
From: Francisco <femolina@uc.cl>
Date: Tue, 15 Nov 2016 11:17:07 -0300
Subject: [PATCH] added missing library

---
 src/mac/LoRaMac.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mac/LoRaMac.h b/src/mac/LoRaMac.h
index 26b7313..fa79164 100644
--- a/src/mac/LoRaMac.h
+++ b/src/mac/LoRaMac.h
@@ -48,6 +48,7 @@
 
 // Includes board dependent definitions such as channels frequencies
 #include "LoRaMac-definitions.h"
+#include "loramac/timer.h"
 
 /*!
  * Beacon interval in ms
-- 
2.7.4


